### ISSUE TITLE
Accessibility of d3 charts for screen readers

### ISSUE THREAD

**Issue by ryelle** [woocommerce/wc-admin#208]
Currently, the store performance charts are not very screen-reader friendly. In this video, I let VoiceOver just read the chart, and it's incomprehensible:

https://cloudup.com/cUtIkC96zHL

There's also no way to trigger the hover cards using just my keyboard.

[This chart is a great example of an accessible interface](https://www.highcharts.com/samples/highcharts/accessibility/advanced-accessible/), but it's also using a different framework, and has a lot of chart-specific copy to inform screen reader users of the content of the chart in plain english - all of this is visually hidden:

![screen shot 2018-07-13 at 5 03 47 pm](https://user-images.githubusercontent.com/541093/42713807-bcc60b1a-86be-11e8-857c-0d174c09504f.png)

[This is a demo of me using VoiceOver to navigate into that chart](https://cloudup.com/cxvTwyBB4nA) (ignore about 0:15-0:30, where I fumbled with VoiceOver a bit).

While we probably can't get there immediately, here are a few possible suggestions from bare-minimum to ideal solutions (knowing that the ideal is complex).

- Hide the charts with `aria-hidden`, which is an OK solution when the chart + table are on the same page
- Add tab-index to the data points, and trigger the hover-cards on focus
- Add aria descriptions to each axis, so VO doesn't read out each mark as a separate point
- Add an aria-label or description to each point on the chart, so a screen reader user can navigate between points
- In line charts, group points in [`region`s](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/region.html) per line, so you can navigate between lines, move VO into a line, and navigate the points in a line
- For the bar charts, group each day in a region, so you can navigate between days, move VO into a day, and navigate the values in a single day

**Comment by timmyc** [comment]
Added to the master task list for d3 #164

**Comment by greenafrican** [comment]
Including `aria-selected` here from https://github.com/woocommerce/wc-admin/pull/270#pullrequestreview-144982328

**Comment by ryelle** [comment]
The chart itself should use a heading, not a `span.woocommerce-chart__title`. I'd suggest [using `H`](https://github.com/woocommerce/wc-admin/tree/master/client/layout#section-and-h) so that the chart uses a context-aware heading.

**Comment by Aljullu** [comment]
@ryelle do you think everything listed in this issue has been fixed in #421? In that case I would close this one and, as discussed at https://github.com/woocommerce/wc-admin/pull/421#pullrequestreview-158199764, created a follow-up issue to improve the labeling.

**Comment by ryelle** [comment]
Yeah, I think the chart is generally more accessible now, which was the goal of this issue. We can create any follow-up issues for specific tasks if needed.

The two remaining tasks from this round of work would be the labelling, and [updating the chart header](https://github.com/woocommerce/wc-admin/issues/208#issuecomment-414408391) to use a real header `H` ([that's here in the code](https://github.com/woocommerce/wc-admin/blob/master/client/components/chart/index.js#L222)).

**Comment by Aljullu** [comment]
Great, I will take care of the header issue today or tomorrow and will close this issue once that is merged. Then we can handle the labelling in a follow-up issue/PR.

**Comment by Aljullu** [comment]
With #399, #421 and #480 merged now, I will close this issue. We can create a follow-up in the future to improve labeling.
